### DISCUSSION TITLE
Unable to log in when copying .db file to another machine

### DISCUSSION THREAD

**Discussion by MrSuttonmann** [keystonejs/keystone#7915] [Questions] [answered]
Hi, I have a question around the out-of-the-box auth configuration.

We're planning on using the SQLite configuration, and checking the DB file into source control so it can be deployed to our development, staging and production environments.

I've noticed during development that if I copy the file to another machine, I can't log into Keystone on that instance, even though the `sessionSecret` in `auth.ts` is set to the same value. To fix the issue, I have to manually delete the user out of the database and then re-create it through the admin interface.

I would expect the same `sessionSecret` would allow me to log in? I've even tried copying the entire folder to the other machine but I still get the same behaviour. Is this a bug, or expected behaviour?

**Comment by dcousens** [comment, marked as answer]
You describe the problem as being unable to login, do you mean that literally or in the sense that you are expecting to be able to copy the cookies in your browser between running instances?

For example, if you mean literally, presuming you are following the [`auth` example](https://github.com/keystonejs/keystone/blob/main/examples/auth) and you have a `User` list with `email` (as the `identityField`) and `password` (as the `secretField`), akin to https://github.com/keystonejs/keystone/blob/77789541a3ca82c38a61d218620e320e323ac0dc/examples/auth/schema.ts#L24-L28

Then every time you send a `authenticateUserWithPassword` query with a valid `email` and `password` pair, a valid session cookie should be returned.
It should work.

**Reply by MrSuttonmann** [reply]
I mean literally, as in I'm unable to login with that user in the admin frontend. I'm not copying any cookies as I wouldn't expect to have to.

I may be mistaken that `sessionSecret` is driving the password generation/validation - if that is the case, is there another configurable value I can set so this will work across multiple machines?

**Reply by dcousens** [reply]
`sessionSecret` is only used for the `Iron.seal` encrypted cookie.
During password validation, [nothing configurable is used by the `password` field](https://github.com/keystonejs/keystone/blob/c3a3789c6e4a1761a24cd02904b7549a62113363/packages/core/src/fields/types/password/index.ts), even the `bcrypt` work factor should be determined from the database field.

**Reply by dcousens** [reply]
@MrSuttonmann maybe you could share more of your configuration? I can't think of why this wouldn't be working as described.

**Reply by MrSuttonmann** [reply]
Hmm, interesting.

As for sharing config, we've the same User schema as what you get out-of-the-box, minus the relationship to Posts.

I've managed to replicate this dead easily:

1. Using `yarn create keystone-app` create two new instances in separate folders.
2. Start one instance using `yarn dev`.
3. Create a user through the admin console, then shut the instance down.
4. Copy the `keystone.db` file to the other instance's folder.
5. Start the second instance, and open the admin console in an incognito tab.
6. Try to log in, and notice authentication fails.

I've attached a ZIP with my test `keystone.db` file. You could drop this into a newly created instance instead and try to login with

User: `test`
Password: `testtest`

[keystone.zip](https://github.com/keystonejs/keystone/files/9549008/keystone.zip)

**Reply by MrSuttonmann** [reply]
@dcousens Just wondering if you'd had a chance to look at this? :)

**Reply by dcousens** [reply]
Sorry @MrSuttonmann this dropped off my radar somehow :blue_heart:, I'm now tracking this. Thanks for the reminder

**Reply by dcousens** [reply]
I can't reproduce this, my steps:

- `yarn create keystone-app` ... `first`
- `yarn create keystone-app` ... `second`
- `cd first && yarn dev`
- `http://localhost:3000`, ... name `test`, email `test`, password `testtest`, 
- Success
- `^C` the `yarn dev`, close Firefox

...

- `cp keystone.db ../second`
- `cd ../second && yarn dev`
- `http://localhost:3000` (private window), ... email `test`, password `testtest`
- Success

**Reply by dcousens** [reply]
@MrSuttonmann if you logout of your first instance upon doing the `/init`, then open a new private tab, can you actually use the credentials to login to the first instance?

**Reply by dcousens** [reply]
@MrSuttonmann opening your database attachment

```bash
~ sqlite3 keystone.db

SQLite version 3.39.3 2022-09-05 11:02:23
Enter ".help" for usage hints.
sqlite> SELECT * FROM User;
cl7yv2wid0054psf0mmkk8tnq|test|test@test.com|$2a$10$CwFaMJ6qVnT6k35zLaImsun/ZtTTeKE1ySAFkU22g8Ed4AJ/MMFW.
```

Emphasis `test@test.com` is the email you provided

**Reply by flexdinesh** [reply]
I tested this out as well.

1. Two separate `yarn create keystone-app` and copied one db to the other and was able to login using the creds created in the first app. 👍 
2. Copied the `keystone.db` file @MrSuttonmann attached in this thread to an app created in my local and was able to login using the creds `test@test.com` and `testtest`. 👍 

@MrSuttonmann copying database between environments should work as expected as long as you're using the same email+password in both the environments.

**Reply by MrSuttonmann** [reply]
@dcousens @flexdinesh I'm really sorry for wasting your time. 

I haven't yet confirmed this with my local instance but I have a sinking feeling I was trying to log in with the user's name, rather than the user's email address. 

Imposter syndrome hits hard sometimes!

I'll confirm and then let you know soon.

**Reply by MrSuttonmann** [reply]
@dcousens This indeed was what I was doing. My apologies.

**Reply by dcousens** [reply]
Not a problem @MrSuttonmann,  this happens to all of us.  I can't count how many times I have had a similar experience. Thanks for making the time to report the issue,  it could have been a bug! :blue_heart:
